Sources/HTTP/Models/Client/HTTP+Client.swift

```diff
@@ -1,79 +1,57 @@
 import Transport
+import Sockets
 
 public enum ClientError: Swift.Error {
     case invalidRequestHost
     case invalidRequestScheme
     case invalidRequestPort
     case unableToConnect
     case userInfoNotAllowedOnHTTP
+    case missingHost
 }
 
-public typealias BasicClient = Client<TCPClientStream, Serializer<Request>, Parser<Response>>
-
-public final class Client<
-    ClientStreamType: ClientStream,
-    SerializerType: TransferSerializer,
-    ParserType: TransferParser
->: ClientProtocol
-where
-    ParserType.MessageType == Response,
-    SerializerType.MessageType == Request
+public typealias TCPClient = BasicClient<
+    TCPInternetSocket,
+    Serializer<Request, StreamBuffer<TCPInternetSocket>>,
+    Parser<Response, StreamBuffer<TCPInternetSocket>>
+>
+
+import TLS
+
+public typealias TLSTCPClient = BasicClient<
```

Comment:
In vapor it will be `EngineClient` and `EngineServer` which I think is more descriptive than `BasicClient`. (`EngineClient` and `FoundationClient` make sense)

As for `TCPClient`, it's a client that uses a `TCPInternetSocket` 
For `TLSClient` it's a client that uses a `TLS.InternetSocket` 

With the changes in this PR, clients no longer rely on supporting a `SecurityLayer`. They can be initialized with any stream that conforms to the simple protocols:

Client:
- DuplexStream (read/write)
- ClientStream (connect)
- InternetStream (scheme, host, port)

Server:
- DuplexStream (read/write)
- ServerStream (bind, listen,accept)
- InternetStream (scheme, host, port)

This approach is more bottom up which I think will allow us to stay on the 2.0 tag for the foreseeable future (even as vapor and other things add new features, e.g., some new TLS security layer, etc)